File: app/src/mainJava/java/com.playground/ui/savenote/SaveNoteViewModelFactory.java
package com.playground.ui.savenote;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import com.playground.data.RepositoryNotes;

/**
 * Created by emil.ivanov on 9/9/18.
 */
public class SaveNoteViewModelFactory implements ViewModelProvider.Factory {

    private RepositoryNotes repository;


    public SaveNoteViewModelFactory(RepositoryNotes repositoryNotes) {
        this.repository = repositoryNotes;

    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new SaveNoteViewModel(repository);
    }
}

File: app/src/mainKotlin/kotlin/com.playground/ui/savenote/SaveNoteViewModelFactory.kt
package com.playground.ui.savenote

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import com.playground.data.RepositoryNotes

/**
 * Created by emil.ivanov on 9/9/18.
 */
class SaveNoteViewModelFactory(val repository: RepositoryNotes) : ViewModelProvider.Factory {


    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(SaveNoteViewModel::class.java)) {
            return SaveNoteViewModel(repository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }

}
